pkg/justinstall/registry.go

```diff
@@ -235,16 +239,28 @@ func (e *RegistryEntry) JustInstall(force bool) {
 	e.CreateShims()
 }
 
-func (e *RegistryEntry) installerURL(arch string) string {
+func (e *RegistryEntry) installerURL(arch string) (string, error) {
 	var url string
 
-	if arch == "x86_64" && e.Installer.X86_64 != "" {
-		url = e.Installer.X86_64
+	if arch == "x86_64" {
+		if e.Installer.X86_64 != "" {
+			url = e.Installer.X86_64
+		} else if e.Installer.X86 != "" {
+			url = e.Installer.X86
+		} else {
+			return "", errors.New("No fallback 32-bit download")
+		}
+	} else if arch == "x86" {
+		if e.Installer.X86 != "" {
```

Comment:
Should we turn X86 into a pointer so we can check against nil instead?

Reply:
> I call it my billion-dollar mistake. It was the invention of the null reference in 1965. At that time, I was designing the first comprehensive type system for references in an object oriented language (ALGOL W). My goal was to ensure that all use of references should be absolutely safe, with checking performed automatically by the compiler. But I couldn't resist the temptation to put in a null reference, simply because it was so easy to implement. This has led to innumerable errors, vulnerabilities, and system crashes, which have probably caused a billion dollars of pain and damage in the last forty years.

I'm not sure I see the advantage of turning it into a pointer when both nil and the zero value (empty string) both signal a missing field.